File: Bidding/ClientApp/src/app/auctions/models/add/add-auction-request.model.ts
import { AboutAuctionModel } from "./about-auction.model";
import { AboutAuctionCreatorModel } from "./about-auction-creator.model";
import { VehicleAuctionModel } from "./categories/vehicle-auction.model";
import { ItemAuctionModel } from "./categories/item-auction.model";
import { PropertyAuctionModel } from "./categories/property-auction.model";

export interface AddAuctionRequestModel {
  aboutAuction: AboutAuctionModel;
  vehicleAuction?: VehicleAuctionModel;
  itemAuction?: ItemAuctionModel;
  propertyAuction?: PropertyAuctionModel;
  aboutAuctionCreator: AboutAuctionCreatorModel;
}

File: Bidding/Models/ViewModels/Auctions/Add/AddAuctionRequestModel.cs
using Bidding.Models.ViewModels.Auctions.Shared.Categories;

namespace Bidding.Models.ViewModels.Auctions.Add
{
    public class AddAuctionRequestModel
    {
        public AboutAuctionModel AboutAuction { get; set; }
        public VehicleAuctionModel VehicleAuction { get; set; }
        public ItemAuctionModel ItemAuction { get; set; }
        public PropertyAuctionModel PropertyAuction { get; set; }
        public AboutAuctionCreatorModel AboutAuctionCreator { get; set; }
    }
}
